### PR TITLE
[APP-2475] Implement credit card UI in app games

### PR DESCRIPTION
**Pull request by pedroagostinhoaptoide**
**What does this PR do?**

   - Adds the credit card screen to aptoide games.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] See by commit.

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APP-2475](https://aptoide.atlassian.net/browse/APP-2475)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-2475](https://aptoide.atlassian.net/browse/APP-2475)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[APP-2475]: https://aptoide.atlassian.net/browse/APP-2475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-2475]: https://aptoide.atlassian.net/browse/APP-2475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt:73]: We need this button in the design system to reuse

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt:59]: No need to wrap if we can use `PrimaryButton` directly

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt:403]: Fields text is currently white and not visible...

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: app-games/src/main/res/values/strings.xml
Comment (ricardo-frango) [inline app-games/src/main/res/values/strings.xml:1]: Don't edit the strings file

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/AdyenExceptionMapper.kt
Comment (ricardo-frango) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/AdyenExceptionMapper.kt:52]: Hardcode the strings on this file

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt
Comment (ricardo-frango) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt:183]: this bottom sheet invocation should be included in a single method. it would have two advantages:
1 - you could remove the bottom sheet from the previews
2 - you can create a state in the preview to preview all states

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt
Comment (ricardo-frango) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt:114]: Should use PreviewDark

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt
Comment (ricardo-frango) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt:141]: Should use PreviewLandscapeDark

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt
Comment (ricardo-frango) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt:1]: This file can be deleted.
there is already a PaymentButtonsView in the "feature_payments" package.
Might need to rebase to have that file updated.

**Review by ricardo-frango — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt:223]: If this comment is irrelevant, please remove.
If it is relevant, then what are we missing?

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt:114]: All payments are in light theme, why do we need to preview dark here?

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt
Comment (pedroagostinhoaptoide) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt]: The text button `AppGamesOtherPaymentMethodsButton` is used instead of this. Should it be in the design system?

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt
Comment (pedroagostinhoaptoide) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt:351]: The styles for these fields were added

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/AdyenExceptionMapper.kt
Comment (pedroagostinhoaptoide) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/AdyenExceptionMapper.kt:52]: It might be better for this to be on hold until the PR with all the strings is merged

File: app-games/src/main/res/values/strings.xml
Comment (pedroagostinhoaptoide) [reply app-games/src/main/res/values/strings.xml:1]: It might be better for this to be on hold until the PR with all the strings is merged

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/presentation/PaymentButtonsView.kt]: Of course it should 😄

File: app-games/src/main/res/values/colors.xml
Comment (ricardo-frango) [inline app-games/src/main/res/values/colors.xml:12]: missing new line

File: app-games/src/main/res/values/styles.xml
Comment (ricardo-frango) [inline app-games/src/main/res/values/styles.xml:48]: missing new line

**Review by ricardo-frango — CHANGES_REQUESTED** [review]

**Review by G0retZ — APPROVED** [review]

**Review by ricardo-frango — APPROVED** [review]

**Pull request merged**
